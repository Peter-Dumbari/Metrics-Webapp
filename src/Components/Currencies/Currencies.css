.currencysection_container {
  padding: 0;
  background-color: var(--primary);
  position: relative;
}

.header_container {
  background-color: rgba(0, 0, 0, 0.225);
  padding: 1px 10px;
  color: var(--white);
}

.header {
  font-weight: 500;
}

.currencies {
  padding: 10px;
  display: grid;
  grid-template-columns: auto auto;
  grid-gap: 2px;
}
