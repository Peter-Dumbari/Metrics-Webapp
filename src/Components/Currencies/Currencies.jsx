import React from 'react';
import { useSelector } from 'react-redux';
import CurrencyCard from '../CurrencyCard/CurrencyCard';
import './Currencies.css';

function Currencies() {
  const { currency, filterCurrency } = useSelector((store) => store.currency);

  return (
    <div className="currencysection_container">
      <div className="header_container">
        <h3 className="header">Global Currencies</h3>
      </div>
      <div className="currencies">
        {filterCurrency.length !== 0
          ? Object.entries(filterCurrency).map(([key, value]) => (
            <CurrencyCard key={key} country={value[1]} currency={value[0]} />
          ))
          : Object.entries(currency).map(([key, value]) => (
            <CurrencyCard key={key} country={value} currency={key} />
          ))}
      </div>
    </div>
  );
}

export default Currencies;
