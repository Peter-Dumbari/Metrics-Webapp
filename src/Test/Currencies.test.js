import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Currencies from '../Components/Currencies/Currencies';

jest.mock('react-redux');

describe('Currencies', () => {
  it('matches the snapshot', () => {
    useSelector.mockReturnValue({
      currency: {
        USD: 'United States Dollar',
        EUR: 'Euro',
        GBP: 'British Pound',
      },
      filterCurrency: {},
    });

    const { container } = render(<Currencies />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
