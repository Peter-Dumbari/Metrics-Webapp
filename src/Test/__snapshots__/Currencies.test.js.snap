// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies matches the snapshot 1`] = `
<div
  class="currencysection_container"
>
  <div
    class="header_container"
  >
    <h3
      class="header"
    >
      Global Currencies
    </h3>
  </div>
  <div
    class="currencies"
  />
</div>
`;
